test/white.jl

```diff
@@ -205,24 +205,25 @@ e = [ 5.19346;
       0.28832 ]
 
     atol = 1e-4
-    w_test = WhiteTest(X, e, testtype = :linear)
+    w_test = WhiteTest(X, e, type = :linear)
     @test pvalue(w_test) ≈ 0.1287 atol=atol
 
-    w_test = WhiteTest(X, e, testtype = :linear_and_squares)
+    w_test = WhiteTest(X, e, type = :linear_and_squares)
     @test pvalue(w_test) ≈ 0.2774 atol=atol
 
     w_test = WhiteTest(X, e)
     @test pvalue(w_test) ≈ 0.3458 atol=atol
-
     show(IOBuffer(), w_test)
```

Comment:
Sorry I should have been more specific: this should also check the contents of the `IOBuffer` using e.g. `String(take!(buf)) == """..."""`.

Reply:
@nalimilan Regarding the `eachcol` and Compat. I don't know whether the maintainers like adding another dep. Perhaps this old-fashioned approach could do the trick until support for < Julia 1.1 is dropped?
```
intercept_col = false
for i = 1:K
    col = view(X,:,i)
    intercept_col = first(col) != 0 && all(==(first(col)), col)
    intercept_col && break
end
```

Reply:
@nalimilan Regarding testing the IOBuffer, I could do something like
```
buffer = IOBuffer()
show(buffer, w_test)
str = String(take!(buffer))
@test str == refstr
```
where `refstr` is a predfined string with the expected output. 

But, I see a **problem**: the expected output contains `point estimate:          5.612418635833716`.  That's a lot of digits, and strictly requiring that they all match is somewhat in conflict with the other tests that use `isapprox`. Is there a way around this (other than always rounding the printed results?)

Reply:
You could use `$value` to interpolate the exact value in the string.

Regarding Compat, that's always a tradeoff, but I'd say it's a small and common enough dependency that it's OK. Adding version checks everywhere is precisely what Compat allows to avoid.

Reply:
The latest push includes 
1. a test for show. (I did't find anything similar in this package, so I am not entirely sure what the expectation is.) 
2. a loop for testing if `X` has an intercept term. Not so pretty, but compatible with Julia 1.0.5 and actually pretty fast.